src/Flame/Types.purs

```diff
@@ -38,3 +39,9 @@ foreign import data NodeData :: Type -> Type
 --Html can actually be typed, but since it is only used in FFI code, I don't think it'd be very useful
 -- | The type of virtual nodes
 foreign import data Html :: Type -> Type
+
+--we support events that are not fired on Nothing message
+foreign import messageMapper :: forall message mapped. (Maybe message -> Maybe mapped) -> Html message -> Html mapped
```

Comment:
Do we need Maybe to Maybe mapping? Like, there's a possibility to fire an event from Nothing with such a mapper, and its behaviour seems undefined to me. `(msg -> Maybe msg')` is proper here, however, does it comply to functor laws and nature... 🤔

Reply:
Hmmm I am not sure that would work. `NodeData` is always `message`, because the `Maybe` value is handled by unsafe JavaScript. See `runHandlers` https://github.com/easafe/purescript-flame/blob/97808aa3cca03cdc1da916e8994a4e07b09bfc70/src/Flame/Renderer/Internal/Dom.js#L362

Maybe you can send a PR to this branch?

Reply:
Or we can just double map the value. :laughing:

Reply:
However do you mean? 🤔

Reply:
I mean,
```purs
map f html = messageMapper (map (map f)) html
```
It seems cheap, but should work well.

Actually, it really should be `msg -> Maybe mapped`, but it's not a functor behaviour, so it could be implemented with `messageMapper` separately.

Reply:
Oh wait... I see, we don't need to map it like this. All right, I was talking about Bind behaviour (we can include it, perhaps), but Functor implementation is all right. 

I'll check it on our SVG UI code ASAP.

Reply:
Did you get the chance to @chekoopa ?